### PR TITLE
Update bufr.py for carra2 for local observations

### PR DESCRIPTION
**Pull request by PerDahlgren**
Added features in bufr.py for carra2 to be able to read local observations

### CODE REVIEW

**Comment by trygveasp** [comment]
Can you try to add only your changes? You have removed some of the last developments I think (sigmao etc)

File: pysurfex/bufr.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 """bufr treatment."""
 import logging
+import sys
```
Comment (trygveasp) [inline pysurfex/bufr.py:3]: This is not needed?

File: pysurfex/bufr.py [hunk 2]
```diff
@@ -279,24 +295,29 @@ def __init__(
                         if not exists:
                             logging.debug("Pos does not exist %s %s", pos, var)
                             if var == "relativeHumidityAt2M":
-                                if not np.isnan(t2m) and not np.isnan(td2m):
+                                if not np.isnan(t2m) and not np.isnan(td2m) and np.isnan(rh2m):
                                     try:
                                         value = self.td2rh(td2m, t2m)
-                                        value = value * 0.01
+                                        value = value * 0.01
```
Comment (trygveasp) [inline pysurfex/bufr.py:301]: Remove the white-spaces

**Comment by trygveasp** [comment]
The unit tests fail because we now require stationOrSiteName in the bufr file. Is it ok to have such a requirement @PerDahlgren ? Will this key always be existing?

**Comment by PerDahlgren** [comment]
stationOrSiteName will not always be existing, only in some bufr templates.
I am a bit confused because this worked when I ran harmonie, with bufr
input that was a mix of different bufr-templates, some
with stationOrSiteName and some without it.


tir. 16. jan. 2024 kl. 10:14 skrev Trygve Aspelien ***@***.***

**Comment by PerDahlgren** [comment]
The fail, in the test, seems to come from eccodes.codes_get(bufr, key).
In Bufr2json (in Harmonie) this does not seem give a crash, it just continues with the next key.
I see that different eccodes versions are used: 2.20.0 (used by the test), 2.23.0 in my Harmonie experiment
Can that be the reason?

**Comment by joewkr** [comment]
@PerDahlgren, this crash has nothing to do with eccodes' version since when running the testsuite real reading functions of eccodes are replaced by stubs which return values from a dummy BUFR file implemented as a python dictionary (see [conftest.py#L425-L447](https://github.com/metno/pysurfex/blob/ce967d9df81d2b6706f7158e0e4807a510b1d4fb/tests/conftest.py#L425-L447)). Since in your PR you add a new `stationOrSiteName` key to be read from a BUFR file, test fail because it's missing from the dummy 'BUFR' dict.

So, as @trygveasp mentioned, if `stationOrSiteName` is a required key now, then it seems that it should be added to the dictionary in `conftest.py` to get tests working. Or maybe a correct exception should be thrown when trying to access a missing field?

**Pull request merged**